### PR TITLE
Fix inactivity score calculation

### PR DESCRIPTION
**Pull request by ajsutton**
## PR Description
Fix a bug found by the alpha.6 reference tests in how we calculate inactivity scores.

Also do the prep work for new test types coming in future releases and improve error messages.  Can't actually update to alpha.6 yet because there's a bug in some of the new transition tests so will need to wait for alpha.7.

## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

**Comment by ajsutton** [comment]
Turns out the tests haven't been released yet.  Will reopen when they're available.

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/altair/statetransition/epoch/EpochProcessorAltair.java [hunk 1]
```diff
@@ -123,18 +123,20 @@ public void processInactivityUpdates(
 
       // Increase inactivity score of inactive validators
       final UInt64 currentScore = inactivityScores.getElement(i);
+      UInt64 newScore = currentScore;
       if (validatorStatus.isNotSlashed() && validatorStatus.isPreviousEpochTargetAttester()) {
         if (currentScore.isGreaterThan(0)) {
-          inactivityScores.setElement(i, currentScore.decrement());
+          newScore = currentScore.decrement();
         }
```
Comment (rolfyone) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/altair/statetransition/epoch/EpochProcessorAltair.java]: you dont want to just use minusMinZero here?

Comment (ajsutton) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/altair/statetransition/epoch/EpochProcessorAltair.java]: Probably makes sense to now that it's now updates the list straight away.  Didn't want to have to write to the list if the value hadn't changed.

**Review by rolfyone — APPROVED** [review]
LGTM

**Pull request merged**